currency/storage.go

```diff
@@ -48,142 +78,132 @@ func (s *Storage) SetDefaults() {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+		if settings.ForexProviders[i].Name != DefaultForexProviderExchangeRatesAPI {
+			if settings.ForexProviders[i].APIKey == "" || settings.ForexProviders[i].APIKey == "Key" {
+				log.Warnf(log.Currency, "%s forex provider API key not set, disabling. Please set this in your config.json file\n",
+					settings.ForexProviders[i].Name)
+				continue
 			}
 
-		case "OpenExchangeRates":
-			if overrides.FxOpenExchangeRates ||
-				settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+			if settings.ForexProviders[i].APIKeyLvl == -1 && settings.ForexProviders[i].Name != "ExchangeRates" {
+				log.Warnf(log.Currency, "%s APIKey level not set, functionality is limited. Please review this in your config.json file\n",
+					settings.ForexProviders[i].Name)
 			}
+		}
 
-		case "ExchangeRates":
-			// TODO ADD OVERRIDE
-			if settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+		if settings.ForexProviders[i].PrimaryProvider {
+			if primaryProvider {
+				log.Warnf(log.Currency, "%s disabling primary provider, multiple primarys found. Please review providers in your config.json file\n",
+					settings.ForexProviders[i].Name)
+				settings.ForexProviders[i].PrimaryProvider = false
+			} else {
+				primaryProvider = true
 			}
+		}
+		fxSettings = append(fxSettings, base.Settings(settings.ForexProviders[i]))
+	}
 
-		case "ExchangeRateHost":
-			if overrides.FxExchangeRateHost || settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+	if len(fxSettings) == 0 {
+		log.Warnln(log.Currency, "No foreign exchange providers enabled, setting default provider....")
+		for x := range settings.ForexProviders {
+			if settings.ForexProviders[x].Name != DefaultForexProviderExchangeRatesAPI {
+				continue
 			}
+			settings.ForexProviders[x].Enabled = true
+			settings.ForexProviders[x].PrimaryProvider = true
+			primaryProvider = true
+			log.Warnf(log.Currency, "No valid foreign exchange providers configured. Defaulting to %s.", DefaultForexProviderExchangeRatesAPI)
+			fxSettings = append(fxSettings, base.Settings(settings.ForexProviders[x]))
 		}
 	}
 
-	if len(fxSettings) != 0 {
-		var err error
-		s.fiatExchangeMarkets, err = forexprovider.StartFXService(fxSettings)
-		if err != nil {
-			s.mtx.Unlock()
-			return err
+	if !primaryProvider {
+		for x := range fxSettings {
+			fxSettings[x].PrimaryProvider = true
+			log.Warnf(log.Currency, "No primary foreign exchange provider set. Defaulting to %s.", fxSettings[x].Name)
 		}
+	}
```

Comment:
This seems like it would be more appropriate to set fxSettings[0] as primary and not everything
```
[WARN]  | CURRENCY |  14/02/2022 11:18:19  | No primary foreign exchange provider set. Defaulting to ExchangeRates.
[WARN]  | CURRENCY |  14/02/2022 11:18:25  | No primary foreign exchange provider set. Defaulting to ExchangeRateHost.
```